kong/api/crud_helpers.lua

```diff
@@ -5,16 +5,43 @@ local app_helpers = require "lapis.application"
 
 local _M = {}
 
+--- Will look up a value in the dao.
+-- Either by `id` field or by the field named by 'alternate_field'. If the value is NOT
+-- a uuid, then by the 'alternate_field'. If it is a uuid then it will first try
+-- the `id` field, if that doesn't yield anything it will try again with the
+-- 'alternate_field'.
+-- @param dao the specific dao to search
+-- @param filter filter table to use, tries will add to this table
+-- @param value the value to look up
+-- @param alternate_field the field to use if it is not a uuid, or not found in `id`
+function _M.find_by_id_or_field(dao, filter, value, alternate_field)
+  filter = filter or {}
+  local is_uuid = utils.is_valid_uuid(value)
+  filter[is_uuid and "id" or alternate_field] = value
+
+  local rows, err = dao:find_all(filter)
+  if err then
+    return nil, err
+  elseif is_uuid and not next(rows) then
```

Comment:
yep, same. style + good practice + consistency with other uses (probably worth a cleanup PR to remove that anti-pattern throughout the codebase).